### PR TITLE
Dict updates

### PR DESCRIPTION
**Pull request by elshize**
Providing commands for updating dictionaries and removing old versions of dictionaries in an external script.

Added a newly available dictionary: en-AU

Solves #3166

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3243)

### CODE REVIEW

**Comment by elshize** [comment]
I'm having difficulty understanding the failed tests. Or rather, I'm running `tox -e py36-pyqt58 -- tests/end2end/features/test_qutescheme_bdd.py` and it fails, even on `master`, while it didn't on Travis. Are these supposed to be skipped? I'm a bit confused. Does this have something to do with the fact that webengine is now default?

File: qutebrowser/browser/webengine/spell.py [hunk 1]
```diff
@@ -21,26 +21,46 @@
 
 import glob
 import os
+import re
 
+from qutebrowser.utils import log
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/spell.py]: Please put this after the PyQt5 import with a blank line in between, so it's grouped in stdlib/other/qutebrowser imports (see [pep8](https://www.python.org/dev/peps/pep-0008/#imports))

File: qutebrowser/browser/webengine/spell.py [hunk 2]
```diff
@@ -21,26 +21,46 @@
...
 from PyQt5.QtCore import QLibraryInfo
 
 
+def version(filename):
+    """Extract the version number from the dictionary file name."""
+    version_re = re.compile(r"""
+        .+-(?P<version>[0-9]+-[0-9]+?)\.bdic
+    """, re.VERBOSE)
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/spell.py]: Seems fine to put this in one line: `version_re = re.compile(r".+-(?P<version>[0-9]+-[0-9]+?)\.bdic")`

File: qutebrowser/browser/webengine/spell.py [hunk 3]
```diff
@@ -21,26 +21,46 @@
...
+    match = version_re.match(filename)
+    assert match is not None, \
+        'the given dictionary file name is malformed: {}'.format(filename)
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/spell.py]: I'd prefer an actual check here:

```python
    if match is None:
        raise ValueError('Malformed dictionary file name: {}'
                         .format(filename))
```

File: qutebrowser/browser/webengine/webenginesettings.py [hunk 4]
```diff
@@ -134,12 +134,12 @@ def __init__(self):
         super().__init__('setSpellCheckLanguages', default=[])
 
     def _find_installed(self, code):
-        installed_file = spell.installed_file(code)
-        if not installed_file:
+        local_filename = spell.local_filename(code)
+        if not local_filename:
             message.warning(
-                "Language {} is not installed - see scripts/install_dict.py "
+                "Language {} is not installed - see scripts/dict.py "
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/webenginesettings.py]: `dictcli.py`?

File: scripts/dictcli.py [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+def main():
+    if configdata.DATA is None:
+        configdata.init()
+    parser = get_argparser()
+    argv = sys.argv[1:]
+    args = parser.parse_args(argv)
+    languages = available_languages()
+    if args.cmd is None:
+        parser.print_usage()
+        exit(1)
```
Comment (The-Compiler) [inline scripts/dictcli.py]: Please use `sys.exit`, the builtin `exit` is intended for the Python interactive REPL/shell.

File: scripts/dictcli.py [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+def install(languages):
+    """Install languages."""
+    for lang in languages:
+        try:
+            print('Installing {}: {}'.format(lang.code, lang.name))
+            install_lang(lang)
+        except PermissionError as e:
+            print(e)
+            sys.exit(1)
```
Comment (The-Compiler) [inline scripts/dictcli.py]: You can just `sys.exit(str(e))` here.

File: scripts/dictcli.py [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+
+
+def main():
+    if configdata.DATA is None:
+        configdata.init()
+    parser = get_argparser()
+    argv = sys.argv[1:]
+    args = parser.parse_args(argv)
+    languages = available_languages()
+    if args.cmd is None:
```
Comment (The-Compiler) [inline scripts/dictcli.py]: Hmm, I wonder if we can't make argparse check for this?

File: scripts/dictcli.py [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+        exit(1)
+    elif args.cmd == 'list':
+        print_list(languages)
+    elif args.cmd == 'update':
+        update(languages)
+    elif args.cmd == 'remove-old':
+        remove_old(languages)
+    elif not args.language:
+        print('You must provide a list of languages to install.')
+        exit(1)
```
Comment (The-Compiler) [inline scripts/dictcli.py]: You can do `sys.exit('You must provide a list of languages to install.')` here.

File: qutebrowser/browser/webengine/spell.py [hunk 9]
```diff
@@ -21,26 +21,46 @@
...
+    return [int(n) for n in match.group('version').split('-')]
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/spell.py]: Hmm, semantically it'd make more sense to make this a tuple I think.

**Comment by The-Compiler** [comment]
Can't say more about failing tests without seeing the output you got :wink: Note that you'll need to run with `--qute-bdd-webengine` to run the tests with QtWebEngine though.

**Comment by The-Compiler** [comment]
Looks good - thank you!

**Comment by cincodenada** [comment]
I pulled down the latest changes and went to find the `install_dict.py` scripts per documentation, and then after some hunting found this, hurrah! I've opened a PR (#3255) to update the documentation to reflect the new script, presuming it's what should be used moving forward.

**Pull request merged**